packages/thirdweb/src/indexer/endpoints/getEvents.ts

```diff
@@ -0,0 +1,108 @@
+import type { DecodeEventLogReturnType, Log } from "viem";
+import type { ThirdwebContract } from "../../contract/contract.js";
+import { getClientFetch } from "../../utils/fetch.js";
+import type { Prettify } from "../../utils/type-utils.js";
+import { formatIndexerEvents } from "../formatter.js";
+import { addRequestPagination } from "../paging.js";
+import type {
+  IndexerInternalEvent,
+  IndexerPagingParams,
+  IndexerResponse,
+} from "../types.js";
+import { getEventsEndpoint } from "../urls.js";
+
+export type GetEventsParams = Prettify<
+  {
+    /**
+     * Contract to fetch events for
+     */
+    contract: ThirdwebContract;
+    /**
+     * Start point for events date range. Default is 7 days ago
+     */
+    startDate?: Date;
+    /**
+     * End point for events date range. Default is current date
+     */
+    endDate?: Date;
+  } & IndexerPagingParams
+>;
+
+export type GetEventsResult = {
+  events: Prettify<
+    Log &
+      DecodeEventLogReturnType & {
+        chainId?: number;
+        count: bigint;
+        time?: Date;
+      }
+  >[];
+};
+
+/**
+ * @beta
+ *
+ * Get events emitted by contract(s)
+ *
+ * @param {GetEventsParams} params
+ * @returns {Promise<GetEventsResult>}
+ *
+ * @example
+ * ```ts
+ * import { createThirdwebClient, getEvents, defineChain, getContract } from "thirdweb";
+ *
+ * const startDate = new Date(Date.now() - 48 * 60 * 60_000);
+ * const endDate = new Date();
+ * const contract = getContract({
+ *  address: "0x...",
+ *  client,
+ *  chain: defineChain(1)
+ * });
+ * const events = await getEvents({
+ *  contract,
+ *  startDate,
+ *  endDate,
+ *  pageSize: 20,
+ *  page: 1
+ * });
+ * ```
+ */
+export async function getEvents(
```

Comment:
i think if you call this inside getContractEvents() and the formats match it should do the decoding for you